process/sync/baseSync.go

```diff
@@ -846,9 +859,22 @@ func (boot *baseBootstrap) isForcedFork() bool {
 }
 
 func (boot *baseBootstrap) rollBackOnForcedFork() {
-	err := boot.rollBack(false)
-	if err != nil {
-		log.Info(err.Error())
+	for {
+		currHeader, err := boot.blockBootstrapper.getCurrHeader()
+		if err != nil {
+			log.Info(err.Error())
+			break
+		}
+
+		if currHeader.GetNonce() <= boot.forkDetector.GetHighestFinalBlockNonce() {
+			break
+		}
+
+		err = boot.rollBack(false)
```

Comment:
so in case of forced fork, we do not add headers to blacklist ? shouldn't add them in any case ?

Reply:
They were somewhat validated by interceptor.

Reply:
Actually I would add in blacklist only headers which are clearly constructed on the fork (for shards only them which are rollback because of the fork triggered by meta). In this way rollback which is triggered because of repeatedly time out error, or because the consensus could not advance, or so on, and not because of a real fork would not be added to blacklist.

Reply:
Actually I would add in blacklist only headers which are clearly constructed on the fork (for shards only them which are rollback because of the fork triggered by meta). In this way rollback which is triggered because of repeatedly time out error, or because the consensus could not advance, or so on, and not because of a real fork would not be added to blacklist.

Reply:
Actually I would add in blacklist only headers which are clearly constructed on the fork (for shards only them which are rollback because of the fork triggered by meta). In this way rollback which is triggered because of repeatedly time out error, or because the consensus could not advance, or so on, and not because of a real fork would not be added to blacklist.